### ISSUE TITLE
GTD not working on certain inaccessible members

### ISSUE THREAD

**Issue by amcasey** [dotnet/roslyn#4914]
I'm seeing some really weird behavior while refactoring.  It looks like GTD is failing because the semantic model is returning 0 candidates even though the candidate reason is Inaccessible.

**Comment by amcasey** [comment]
1. Checkout https://github.com/amcasey/roslyn/tree/Repro4914
 2. Open src\InteractiveWindow\Editor\InteractiveWindow.cs 
 3. GTD on ```Evaluator``` on line 95, ```Copy``` on line 344, ```Write``` on line 448.  Other squiggled (i.e. inaccessible) identifiers seem to work.

```Write``` seems significant since it has an overload and ```WriteLine``` (just below) works correctly.  However, ```Evaluator``` is a field, which can't have an overload.

I debugged with @dpoeschl and we found that the semantic model was returning no candidates, which seems incorrect.
